### PR TITLE
[WOR-761] Azure workspaces should return canCompute as true for Writers

### PR DESCRIPTION
**Pull request by cahrens**
[Ticket](https://broadworkbench.atlassian.net/browse/WOR-761)

Note that there is integration (swatomation) test coverage for the Google case.

---

**PR checklist**

- [ ] Include the JIRA issue number in the PR description and title
- [ ] Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] If you changed anything in `model/`, then you should [publish a new official `rawls-model`](https://github.com/broadinstitute/rawls/blob/develop/README.md#publish-rawls-model) and update `rawls-model` in [Orchestration's dependencies](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/project/Dependencies.scala).
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green, including CI tests
- [ ] **Squash commits and merge** to develop (branches are automatically deleted after merging)
- [ ] Inform other teams of any substantial changes via Slack and/or email

### CODE REVIEW

**Comment by cahrens** [comment]
jenkins retest

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 1]
```diff
@@ -533,9 +533,18 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
...
+                                        userAccessLevel: WorkspaceAccessLevel,
+                                        cloudPlatform: Option[WorkspaceCloudPlatform]
+  ): Future[Boolean] =
+    cloudPlatform match {
+      case Some(WorkspaceCloudPlatform.Azure) =>
+        Future.successful(userAccessLevel >= WorkspaceAccessLevels.Write)
+      case _ if userAccessLevel >= WorkspaceAccessLevels.Owner =>
+        Future.successful(true)
+      case default =>
+        samDAO.userHasAction(SamResourceTypeNames.workspace, workspaceId, SamWorkspaceActions.compute, ctx)
```
Comment (cahrens) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala:546]: This is covered by integration tests (WorkspaceApiSpec).

**Review by aherbst-broad — APPROVED** [review]

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 2]
```diff
@@ -533,9 +533,18 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
...
+  private def getUserComputePermissions(workspaceId: String,
+                                        userAccessLevel: WorkspaceAccessLevel,
+                                        cloudPlatform: Option[WorkspaceCloudPlatform]
+  ): Future[Boolean] =
+    cloudPlatform match {
+      case Some(WorkspaceCloudPlatform.Azure) =>
+        Future.successful(userAccessLevel >= WorkspaceAccessLevels.Write)
+      case _ if userAccessLevel >= WorkspaceAccessLevels.Owner =>
+        Future.successful(true)
+      case default =>
```
Comment (marctalbott) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala:545]: nit: naming this `default` made me look for where we use `default`. I generally prefer `_` when the value isn't used later

**Review by marctalbott — APPROVED** [review]

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala (hunk 2 above)
Comment (cahrens) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala:545]: Yeah, I went back and forth on this. I like the clarity of "default" for saying "this is the default case if nothing else fits", but I realize that's probably a bit pedantic. In any case, I have no desire to roll the dice again with swatomation.

Comment (cahrens) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala:545]: Ugh, forgot this repo doesn't auto-squash commits and hit the merge button too fast. :(

**Pull request merged**